lib/Dialect/FIRRTL/Transforms/IMConstProp.cpp

```diff
@@ -500,9 +500,15 @@ void IMConstPropPass::visitConnect(ConnectOp connect) {
   // Driving result ports propagates the value to each instance using the
   // module.
   if (auto blockArg = connect.dest().dyn_cast<BlockArgument>()) {
-    if (!AnnotationSet::get(blockArg).hasDontTouch())
-      for (auto userOfResultPort : resultPortToInstanceResultMapping[blockArg])
-        mergeLatticeValue(userOfResultPort, srcValue);
+    // If the port is marked DontTouch, we cannot propogate any value.
+    if (!srcValue.isOverdefined() &&
+        AnnotationSet::get(blockArg).hasDontTouch())
+      srcValue = LatticeValue::getOverdefined();
```

Comment:
Does the other patch subsume this?  That seems like a better way to go.

Reply:
yes